Clean lwaftr object files

dpino opened the pull request:
Fixes #678 cc @takikawa.

kbara commented on src/Makefile:
```diff
@@ -209,7 +209,14 @@ obj/doc/snabb.epub: obj/doc/snabb.markdown
-clean:
+clean_programs:
+	@(for d in program/*/; do \
+		if [[ -f "$$d/Makefile" ]]; then \
```
``` 
% make clean_programs
/bin/sh: 2: [[: not found
```

dpino committed:
Execute make clean in programs containing a Makefile

- program/lwaftr/Makefile: Clean up .o files.
- program/snabbvmx/Makefile: Clean up .o files.

dpino commented:
Fixed comment. I reworked the script a bit. Example of execution:

```
$ make clean
CLEAN     program/lwaftr/
RM        ./tests/data/no_icmp.o
RM        ./tests/data/icmp_on_fail.o
RM        ./tests/data/vlan.o
RM        doc/snabb-lwaftr.*
CLEAN     program/snabbvmx/
RM        doc/snabbvmx.*
RM        snabb obj bin testlog programs.inc
```

kbara commented:
That's looking a lot better, but snabbvmx's Makefile needs better cleaning:

```
% find program -name '*.o'
program/snabbvmx/tests/end-to-end/data/binding_table.txt.s.o
program/snabbvmx/tests/end-to-end/data/snabbvmx-lwaftr-xe1.o
program/snabbvmx/tests/end-to-end/data/vlan/binding_table.txt.s.o
program/snabbvmx/tests/end-to-end/data/vlan/snabbvmx-lwaftr-xe1.o
```

dpino commented:
OK, I added the a clean_object_files rule in `snabbvmx/Makefile`. Ideally the patch should be ported to `lwaftr-snabbvmx` but while lwaftr-snabbvmx doesn't catch up with lwaftr branch, I would only do that for critical bugs. This is more like an enhancement.

teknico reviewed (approved):
LGTM

kbara commented:
LGTM too. @dpino , will you merge this?
